src/classes/CON_DoNotContact_TDTM.cls

```diff
@@ -32,95 +32,177 @@
+     * @param triggerAction The current trigger action
+     * @return returns true if trigger context is BeforeUpdate, false otherwise
+     */ 
+    @TestVisible
+    private Boolean isBeforeUpdate(TDTM_Runnable.Action triggerAction) {
+        return triggerAction == TDTM_Runnable.Action.BeforeUpdate;
+    }
 
-        con.Do_Not_Contact__c = con.Deceased__c;
-        markContactDoNotContact(con);
+    /**
+     * @description Execute the logic necessary for a BeforeInsert trigger action.
+     * @param newContactsList List of Contacts inserted
+     */ 
+    @TestVisible
+    private void processBeforeInsert(List<Contact> newContactsList) {
+        List<Contact> deceasedContactsList = new List<Contact>();
+        List<Contact> doNotContactContactsList = new List<Contact>();
+
+        for (Contact con : newContactsList){
+            if (con.Deceased__c == true){
+                con.Do_Not_Contact__c = con.Deceased__c;
+                deceasedContactsList.add(con);
+            }
 
-        con.Exclude_from_Household_Name__c = con.Deceased__c;
-        con.Exclude_from_Household_Formal_Greeting__c = con.Deceased__c;
-        con.Exclude_from_Household_Informal_Greeting__c = con.Deceased__c;
+            if (con.Do_Not_Contact__c == true){
+                doNotContactContactsList.add(con);
+            }
+        }
 
-        copyNamingExclusionsFromCheckboxes(con);
+        if (deceasedContactsList.isEmpty() == false){
+            this.syncNamingExclusionsOnContacts(deceasedContactsList);
+            this.syncDoNotContactFieldsOnContacts(deceasedContactsList);
+        }
+
+        if (doNotContactContactsList.isEmpty() == false){
+            this.syncDoNotContactFieldsOnContacts(doNotContactContactsList);
+        }
     }
 
-	/*******************************************************************************************************
-    * @description Toggles additional Contact fields as necessary when a contact is flagged as "do not contact". 
-    * @param con The Contact record.
-    * @return void.  
-    ********************************************************************************************************/
-    public void markContactDoNotContact(Contact con) {
+    /**
+     * @description Execute the logic necessary for a BeforeUpdate trigger action.
+     * @param newContactsList List of updated contacts
+     * @param oldContactsById Map of contacts' state prior to update
+     */ 
+    @TestVisible
+    private void processBeforeUpdate(List<Contact> newContactsList, Map<Id, Contact> oldContactsById) {
+        List<Contact> deceasedContactsList = new List<Contact>();
+        List<Contact> doNotContactContactsList = new List<Contact>();
+        
+        for (Contact con : newContactsList){
+            Contact oldCon = oldContactsById.get(con.Id);
+
+            if (con.Deceased__c != oldCon.Deceased__c){
+                con.Do_Not_Contact__c = con.Deceased__c;
+                deceasedContactsList.add(con);
```

Comment:
@m-kinkade @MatthewBlanski  This variable name isn't accurate based on what it's doing. It's not necessarily that they've been marked deceased. It's that the deceased field has been changed. As it works now (and as it did before the refactor), unchecking the deceased field would also uncheck all the other fields. Same with the Do Not Contact field. I'm not suggesting we change the functionality as part of this PR but variable names could be more accurate (e.g changedDoNotConactList)